transactor/transactor.go

```diff
@@ -135,21 +136,30 @@ func (t *Transactor) SendTxMsg(msg sdk.Msg) (*sdk.TxResponse, error) {
 }
 
 func (t *Transactor) SendTxMsgs(msgs []sdk.Msg) (*sdk.TxResponse, error) {
-	txBytes, stdSignMsg, err := t.signTx(msgs)
-	if err != nil {
-		return nil, fmt.Errorf("signTx err: %s", err)
-	}
-	txResponse, err := t.CliCtx.BroadcastTx(txBytes)
-	if err != nil {
-		return nil, fmt.Errorf("BroadcastTx err: %s", err)
-	}
+	for try := 0; try < maxTxRetry; try++ {
+		txBytes, stdSignMsg, err := t.signTx(msgs)
+		if err != nil {
+			return nil, fmt.Errorf("signTx err: %s", err)
+		}
+		txResponse, err := t.CliCtx.BroadcastTx(txBytes)
+		if err != nil {
+			return nil, fmt.Errorf("BroadcastTx err: %s", err)
+		}
 
-	if txResponse.Code != sdkerrors.SuccessABCICode {
-		return &txResponse, fmt.Errorf("BroadcastTx failed with code: %d, rawLog: %s, stdSignMsg: %s acct %s accnum %d seq %d",
-			txResponse.Code, txResponse.RawLog, t.Key.GetAddress(), stdSignMsg.ChainID, stdSignMsg.AccountNumber, stdSignMsg.Sequence)
-	}
+		if txResponse.Code == sdkerrors.SuccessABCICode {
+			return &txResponse, nil
+		}
 
-	return &txResponse, nil
+		err = fmt.Errorf("BroadcastTx failed with code: %d, rawLog: %s, stdSignMsg chainId: %s acct: %s accnum: %d seq: %d",
+			txResponse.Code, txResponse.RawLog, stdSignMsg.ChainID, t.Key.GetAddress(), stdSignMsg.AccountNumber, stdSignMsg.Sequence)
+		if txResponse.Code == sdkerrors.ErrUnauthorized.ABCICode() {
+			log.Warnln(err.Error(), "retrying")
+			time.Sleep(txRetryDelay)
+		} else {
+			return &txResponse, err
+		}
+	}
+	return nil, errors.New("Maximum tx retries reached")
```

Comment:
return the "BroadcastTx failed with code:...." err

Reply:
Done.